calculation/sha_calc/sha_calc/directivity/bea20/bea20.py

```diff
@@ -0,0 +1,224 @@
+import numpy as np
+import numpy.matlib
+import math
+from typing import List
+
+
+def bea20(
+    M: float,
+    U: List,
+    T: List,
+    Smax: List,
+    D: float,
+    Tbot: float,
+    Dbot: float,
+    Rake: float,
+    Dip: float,
+    type: int,
+    Period: float,
+):
+    """
+    Calculates the directivity effects at given locations based on fault information.
+    Algorithm has been taken from the Bayless 2020 model found in the paper.
+    "A Rupture Directivity Adjustment Model Applicable to the NGA-West2 Ground Motion Models and Complex Fault Geometries
+    (Bayless 2020)".
+    This functionality was compared to the matlab code taken from the paper.
+
+    Parameters
+    ----------
+    M: float
+        Moment magnitude, 5<=M<=8
+    U: list
+        The GC2 coordinates in km. Equivalent to ry.
+        Must be nX1 number where n is the number of locations at which the model provides a prediction.
+    T: list
+        The GC2 coordinates in km. Equivalent to rx.
+        Must be nX1 number where n is the number of locations at which the model provides a prediction.
+    Smax: List
+        List of 2x1 number where first is the lowest max value and the second is the largest max value for s.
+        Computed from GC2 using nominal strike of the fault.
+    D: float
+        The effective rupture travel width, measured from the hypocenter to the shallowest depth of the rupture plane.
+    Tbot: float
+        The length of the bottom of the rupture plane (projected to the surface) in km.
+    Dbot: float
+        The vertical depth of the bottom of the rupture plane from the ground surface, including Ztor, in km.
+    Rake: float
+        The Rake of the fault
+    Dip: float
+        The Dip of the fault at the hypocenter
+    type: float
+        1 for a strike slip
+        2 for a oblique, reverse or normal
+    Period: float
+        Used for specifying the period for a pSA IM
+    """
+
+    # If not specified, determine rupture category from Rake angle
+    if type == 0:
+        if (-30 <= Rake <= 30) or (-180 <= Rake <= -150) or (150 <= Rake <= 180):
+            type = 1
+        else:
+            type = 2
```

Comment:
Just checking, the comment says `if not specified` however under the `directirvity.py`, we are currently passing 0 for type.
Do you reckon it's good to set the type with default 0?